docs/building-apps/moneygram-access-integration-guide.mdx

```diff
@@ -310,10 +422,57 @@ Code for submitting transactions to Stellar should be developed thoughtfully. Th
 - Set a maximum timebound on the transaction. This ensures that if your transaction is not included in a ledger before the set time, you can reconstruct the transaction with a higher offered fee and submit it again with better chances of inclusion.
 - Resubmit the transaction when you get 504 status codes. 504 status codes are just telling you that your transaction is still pending -- not that it has been canceled or that your request was invalid. You should simply make the request again with the same transaction to get a final status (either included or expired).
 
-Below is highly simplified code for submitting a payment transaction. It does not use timebounds, handle a transaction’s expiration, or handle 504 status codes.
+Below is a sample `Typescript` code for submitting a payment ("transfer") transaction using the Wallet SDK. It uses timebounds and handles 504 status codes (inside `submitTransaction`), but it does not handle a transaction’s expiration.
+
+Please note that the `Python` code is highly simplified. It does not use timebounds, handle 504 status codes or handle a transaction’s expiration.
 
 <CodeExample>
 
+```ts
+import { Wallet, IssuedAssetId } from "@stellar/typescript-wallet-sdk";
+
+const wallet = Wallet.TestNet();
+
+// This creates a Stellar instance to manage the connection with Horizon.
+const stellar = wallet.stellar();
+
+// Creates a stellar asset, in this case it should use USDC configuration.
+const asset = new IssuedAssetId(ASSET_CODE, ASSET_ISSUER);
+
+// This creates a transaction builder which we'll be using to assemble
+// our transfer withdrawal transaction as shown below.
+const txBuilder = await stellar.transaction({
+  sourceAddress: FUNDS_STELLAR_KEYPAIR,
+  baseFee: 10000, // this is 0.001 XLM
+  timebounds: 180, // in seconds
+});
+
+// We can use the transaction object received on the onMessage callback from
+// the watcher, or, we can also fetch the transaction object using either
+// getTransactionBy or getTransactionsForAsset as illustrated in previous step.
+onMessage: (transaction) => {
+  if (transaction.status === "pending_user_transfer_start") {
+    // Use the builder to assemble the transfer transaction. Behind the scenes
+    // it extracts the Stellar account (withdraw_anchor_account), memo (withdraw_memo)
+    // and amount (amount_in) to use in the Stellar payment transaction that will
+    // be submitted to the Stellar network.
+    const transferTransaction = txBuilder
+      .transferWithdrawalTransaction(transaction, asset)
+      .build();
+
+    // Signs it with the source (funds) account key pair
+    transferTransaction.sign(FUNDS_STELLAR_KEYPAIR);
+
+    // Finally submits it to the stellar network. This stellar.submitTransaction()
+    // function handles '504' status codes by keep retrying it until submission
+    // succeeds or we get a different error.
+    const response = await stellar.submitTransaction(transferTransaction);
```

Comment:
Should we write an example on how to handle other errors?

Reply:
https://github.com/stellar/stellar-docs/pull/658/commits/0f45a2dcd0cde262d3489b89fd64785877ef01cc